### PR TITLE
TextInput `numberOfLines`/`maxNumberOfLines` workaround (manually calculating `maxHeight` of TextInput

### PR DESCRIPTION
**Pull request by chrispader**
@Szymon20000 @roryabraham 

### Details

Temporary workaround for `numberOfLines`/`maxNumberOfLines` props on `TextInput` component. This workaround uses `maxHeight` and `lineHeight` styles to simulate the effect of the missing props, until the upstream RN [RFC](https://github.com/react-native-community/discussions-and-proposals/discussions/731) has been approved and the PR was merged.

https://expensify.slack.com/archives/C035J5C9FAP/p1700157933350519?thread_ts=1699609652.957299&cid=C035J5C9FAP

### Fixed Issues

$ 
PROPOSAL: 

### Tests

1. Go to a report/chat
2. Press enter a couple of times in the composer
3. Make sure the composer doesn't grow to more than 6 lines
4. Go into full-size mode
5. Make sure the composer extends to ~50% of the screen
6. Exit full-size mode
7. Clean input
8. Make sure the composer shrinks again

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as in Tests.

### QA Steps

1. Check the same tests in release build

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

[android.webm](https://github.com/Expensify/App/assets/20173411/851655a8-ac21-4052-8963-bedb98e1a643)

Screenshot_20231117_164836

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/20173411/9303f9ed-e3b4-4f41-825d-18c3a3b44292

Simulator Screenshot - iPhone 15 Pro - 2023-11-17 at 16 46 23

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

Screen recordings added for iOS and Android

### CODE REVIEW

**Comment by chrispader** [comment]
Ready for review! 👍  @roryabraham @Szymon20000

**Comment by chrispader** [comment]
We should probably prioritise this PR since it massively reduces Android build time, because once we remove this patch we no longer need to compile Android RN from source. :)

cc @roryabraham @AndrewGable

**Comment by danieldoglas** [comment]
@situchan will review this PR as a C+ so we can move faster!

**Comment by chrispader** [comment]
@situchan do you have some time to review this soon? :) Android build speed gainzz would be awesome to have haha

**Comment by situchan** [comment]
reviewing today

**Comment by situchan** [comment]
install time

👍

File: src/styles/StyleUtils.ts
Comment (situchan) [inline src/styles/StyleUtils.ts:1394]: Please update this comment

**Comment by situchan** [comment]
BUG1: [iOS] last line is not removed after removing all text on that line

https://github.com/Expensify/App/assets/108292595/1c3a512d-d44f-472a-b350-3ef9e9c3e2fc

**Comment by chrispader** [comment]
> BUG1: [iOS] last line is not removed after removing all text on that line

As far as i can see, this bug is only present on simulators. I've tested the current solution on physical devices and it works.

Seems to me, this is a bug in how the simulator renders the TextInput

**Comment by situchan** [comment]
@chrispader conflicts

**Comment by chrispader** [comment]
> @chrispader conflicts

resolved!

**Review by AndrewGable — DISMISSED** [review]

**Comment by situchan** [comment]
Please pull main. It's strange that when I pull main locally, it conflicts but not on github

**Comment by chrispader** [comment]
@situchan main conflicts resolved! :)

**Comment by situchan** [comment]
patch was unremoved. Is that correct?

**Comment by chrispader** [comment]
> patch was unremoved. Is that correct?

Yes i kept it because it contains valid changes for iOS. We're not using it on `Android` though, since we're not building RN from source.

**Comment by situchan** [comment]
Multiline issue still happens on android. No matter patch applied or not

android bug

**Comment by chrispader** [comment]
> Multiline issue still happens on android. No matter patch applied or not

This is not what the initial `numberOfLines`/`maxNumberOfLines` PR was trying to solve, afaik. The changes only affect the `TextInput` in the `Composer` component

**Comment by situchan** [comment]
> > Multiline issue still happens on android. No matter patch applied or not
> 
> This is not what the initial `numberOfLines`/`maxNumberOfLines` PR was trying to solve, afaik. The changes only affect the `TextInput` in the `Composer` component

This happens only on this PR. Not happening on main

**Comment by situchan** [comment]
> I get this behaviour on `main` as well.

Are you sure? I am not able to reproduce.

**Comment by chrispader** [comment]
> > I get this behaviour on `main` as well.
> 
> Are you sure? I am not able to reproduce.

yes, sorry you were right. I forgot that the "number of lines" patch also included JS files from RN. Everything should work now 👍

cc @situchan

**Comment by situchan** [comment]
Thanks. @AndrewGable please generate QR build when you get a chance

**Comment by situchan** [comment]
QR build works well
@chrispader please fix last conflict

File: src/components/Composer/index.native.js
Comment (fabioh8010) [inline src/components/Composer/index.native.js:112]:
```suggestion
            smartInsertDelete={false}
```

Hey @chrispader 👋 , jumped on this out of curiosity, why are we setting `smartInsertDelete` to `true`? It should be `false` in order to fix [this issue](https://github.com/Expensify/App/pull/15051). At the moment isn't doing anything but it will start working when we upgrade the App to 0.73.

**Comment by chrispader** [comment]
@situchan @danieldoglas conflicts resolved again :) Please do a final review by chance and merge

**Comment by situchan** [comment]
I am heard this should be merged after RN upgrade.
Or something changed regarding merge order? cc: @roryabraham

**Comment by danieldoglas** [comment]
Looks like we just merged the RN update https://github.com/Expensify/App/pull/31558.

We should probably merge main into this branch and do a new set of tests @chrispader @hannojg @situchan

**Comment by chrispader** [comment]
> Looks like we just merged the RN update #31558.
> 
> We should probably merge main into this branch and do a new set of tests @chrispader @hannojg @situchan

@danieldoglas done! 👍

Throoughly testing it now :)

**Comment by chrispader** [comment]
Updated screen recordings after recent RN73 upgrade and `main` merge:

**Android**
[android-nol.webm](https://github.com/Expensify/App/assets/20173411/2626bb6e-2b0a-45ba-a9a6-88f9690cc8b7)

**Comment by chrispader** [comment]
@danieldoglas can we merge this? this will speed up the Android build by a lot :)

**Comment by situchan** [comment]
@chrispader please merge main. I will retest shortly today

**Comment by situchan** [comment]
@danieldoglas can we generate adhoc build once more?

**Comment by chrispader** [comment]
> @chrispader please merge main. I will retest shortly today

there were no conflicts, but just updated it!

**Comment by situchan** [comment]
TS check failing

**Comment by chrispader** [comment]
> TS check failing

ah lol, locally it wasn't...

This is exactly why i had this change. `rows` is somehow not included in the `TextInputProps` type. Is this a problem in other PR's as well? Because i can't see how this is related to this one.


Gonna change it back to `numberOfLines` then, because it does the same..

**Comment by chrispader** [comment]
> TS check failing

yea, that's exactly the TS error that i meant, though i have no idea why this is triggered. Is this shown in other PRs as well? Cause i can't see how this i related to this PR...

We remove the patch, but `rows` is still a valid prop on `TextInput` in the vanilla `react-native`

**Comment by chrispader** [comment]
> TS check failing

This only happens in the CLI, locally it works. Can we maybe reset the CLI cache? @danieldoglas @situchan

**Comment by situchan** [comment]
> > TS check failing
> 
> This only happens in the CLI, locally it works.

Also works for me locally. Please try pull main again and see if that fixes

**Comment by danieldoglas** [comment]
Here's the specific error:

```
Type '{ onSelectionChange: (event: NativeSyntheticEvent<TextInputSelectionChangeEventData>) => void; rows: number; disabled: boolean; ... 14 more ...; autoFocus: boolean; }' is not assignable to type 'IntrinsicAttributes & TextInputProps & RefAttributes<Component<AnimatedProps<TextInputProps>, {}, any>>'.
```

I've triggered the execution again and it failed with the same error

**Comment by situchan** [comment]
A few iOS bugs:

- scrolled down automatically while typing and not able to see what is typing

https://github.com/Expensify/App/assets/108292595/0e9a2833-fd96-4f97-b66e-02582d8de3c7

- space between last line and composer bottom border line after scrolling down to bottom


https://github.com/Expensify/App/assets/108292595/17eecfc1-0f01-4d10-8d3f-1aa22e7903b1

**Comment by chrispader** [comment]
> A few iOS bugs:
> 
> * scrolled down automatically while typing and not able to see what is typing
> 
>  Screen.Recording.2024-01-18.at.10.25.14.PM.mov 
> 
> * space between last line and composer bottom border line after scrolling down to bottom
> 
>  Screen.Recording.2024-01-18.at.10.31.38.PM.mov

can you please check if these bugs also occur on a physical device?

We've had similar bugs (and exactly the last one) before and it was an issue with the simulator. If they do not appear on a physical iOS device it's not a bug

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by situchan** [comment]
> > A few iOS bugs:
> > 
> > * scrolled down automatically while typing and not able to see what is typing
> > 
> > Screen.Recording.2024-01-18.at.10.25.14.PM.mov
> > 
> > * space between last line and composer bottom border line after scrolling down to bottom
> > 
> > Screen.Recording.2024-01-18.at.10.31.38.PM.mov
> 
> can you please check if these bugs also occur on a physical device?
> 
> We've had similar bugs (and exactly the last one) before and it was an issue with the simulator. If they do not appear on a physical iOS device it's not a bug

I am not able to install build on physical iPhone for now. Only tested on simulator.

@danieldoglas Not just for this PR but can I register my device for future adhoc builds?
similar to https://github.com/Expensify/App/pull/32505

**Comment by danieldoglas** [comment]
@situchan can you open a request on the contributors channel?

**Comment by chrispader** [comment]
Also for the TS error. I'm pretty sure it is because the CLI caches `node_modules` and the removed patch is still somehow applied, maybe incorrectly. The vanilla RN version includes the `rows` prop on `TextInput` and locally `npm run typecheck succeeds.`

**Comment by chrispader** [comment]
It seems the `rows` prop was removed from the TS declaration file in RN0.73. Can we just replace it with `numberOfLines` since it's evaluating to the same in the source.

https://github.com/facebook/react-native/blob/fd0ca4dd6209d79ac8c93dbffac2e3dca1caeadc/packages/react-native/Libraries/Components/TextInput/TextInput.js#L1557

**Comment by situchan** [comment]
> It seems the `rows` prop was removed from the TS declaration file in RN0.73. Can we just replace it with `numberOfLines` since it's evaluating to the same in the source.
> 
> https://github.com/facebook/react-native/blob/fd0ca4dd6209d79ac8c93dbffac2e3dca1caeadc/packages/react-native/Libraries/Components/TextInput/TextInput.js#L1557

Is it same on react-native-web as well?
Not sure why RNW team replaced `numberOfLines` with `rows` while RN keeps using `numberOfLines`. Seems like no plan yet to deprecate.

**Comment by chrispader** [comment]
> > It seems the `rows` prop was removed from the TS declaration file in RN0.73. Can we just replace it with `numberOfLines` since it's evaluating to the same in the source.
> > https://github.com/facebook/react-native/blob/fd0ca4dd6209d79ac8c93dbffac2e3dca1caeadc/packages/react-native/Libraries/Components/TextInput/TextInput.js#L1557
> 
> Is it same on react-native-web as well? Not sure why RNW team replaced `numberOfLines` with `rows` while RN keeps using `numberOfLines`. Seems like no plan yet to deprecate.

In `react-native-web` `numberOfLines` is deprecated but still functional. We can either ignore the TS error in code and use `rows` or use the deprecated `numberOfLines` prop...

**Comment by mountiny** [comment]
Creating a new test build to see if this will work for @situchan phone

**Comment by situchan** [comment]
I am still getting `This app cannot be installed because its integrity could not be verified.`

**Comment by mountiny** [comment]
I forgot we need to first sync with main

**Comment by situchan** [comment]
@chrispader please sync when you get a chance. Once I confirm they don't happen on real iPhone, we can merge

**Comment by chrispader** [comment]
> @chrispader please sync main when you get a chance. Once I confirm they don't happen on real iPhone, we can merge

done! @mountiny @situchan

**Comment by mountiny** [comment]
triggered

**Comment by mountiny** [comment]
The error is not clear, @chrispader can you sycn with main again, other PRs are passing the build just fine https://github.com/Expensify/App/actions/runs/7626675136/job/20774453791

**Comment by chrispader** [comment]
just merged main and built locally again. @mountiny could you trigger another build here? :)

**Comment by mountiny** [comment]
https://github.com/Expensify/App/actions/runs/7628504108

**Comment by chrispader** [comment]
> 🧪🧪 Use the links below to test this adhoc build on Android, iOS, Desktop, and Web. Happy testing! 🧪🧪
> 
> Android 🤖	iOS 🍎
> https://ad-hoc-expensify-cash.s3.amazonaws.com/android/31483/index.html	❌ FAILED ❌
> Android	The QR code can't be generated, because the iOS build failed
> Desktop 💻	Web 🕸️
> https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/31483/NewExpensify.dmg	https://31483.pr-testing.expensify.com
> Desktop	Web
> 👀 [View the workflow run that generated this build](https://github.com/Expensify/App/actions/runs/7628504108) 👀

checking again.. 😬

**Comment by chrispader** [comment]
@mountiny after running `npm run pod-install` there's a diff for the `Podfile.lock`. Let's try again with this diff?

Also looking into the RN source right now, to try fix the issue...

**Comment by situchan** [comment]
`Yoga` checksum depends on cocoapods version where pod is being installed
`e64aa65de36c0832d04e8c7bd614396c77a80047` in 1.13.0
`13c8ef87792450193e117976337b8527b49e8c03` in 1.14.3

**Comment by chrispader** [comment]
> @mountiny after running `npm run pod-install` there's a diff for the `Podfile.lock`. Let's try again with this diff?
> 
> Also looking into the RN source right now, to try fix the issue...

The error is weirdly coming from `@react-native-firebase/perf`. Though there seems nothing is off with https://github.com/invertase/react-native-firebase/blob/9f00ac9011a710400097302b8263e5c8cf7a6575/packages/perf/ios/RNFBPerf/RNFBPerfModule.m#L18

**Comment by mountiny** [comment]
https://github.com/Expensify/App/actions/runs/7629870160

**Comment by chrispader** [comment]
> https://github.com/Expensify/App/actions/runs/7629870160

now it's working! 🎉

**Comment by chrispader** [comment]
@mountiny can we merge this now that iOS is building? Getting conflicts with `main` and i don't think we're holding on anything anymore? :)

**Comment by situchan** [comment]
@mountiny Let's do that! I couldn't still be able to install above build on my iPhone but should not be blocker.
I will verify once hit staging.

**Comment by mountiny** [comment]
@AndrewGable @danieldoglas can one of you approve and merge since you have been involved in this PR?

**Review by danieldoglas — DISMISSED** [review]

**Comment by danieldoglas** [comment]
K, another conflict on this. I'm online for the next 3 hours, so please ping me if you solve the conflicts in that time and I'll merge it.

**Comment by chrispader** [comment]
> K, another conflict on this. I'm online for the next 3 hours, so please ping me if you solve the conflicts in that time and I'll merge it.

resolved! @danieldoglas

**Review by danieldoglas — APPROVED** [review]

**Comment by danieldoglas** [comment]
Waiting for the checks, then will merge.

**Review by AndrewGable — APPROVED** [review]

File: ios/Podfile.lock
Comment (situchan) [inline ios/Podfile.lock:1970]: NAB: this change is irrelevant to this PR

**Review by situchan — APPROVED** [review]

**Comment by chrispader** [comment]
> @chrispader can you please check this urgently?
> 
> On main, I am seeing multiple lines in LHN.
> 
> Screenshot 2024-01-26 at 2 23 41 PM
> Still works fine on [this](https://github.com/Expensify/App/pull/31483#issuecomment-1906645638) adhoc build.
> 
> Screenshot 2024-01-26 at 2 24 42 PM
> So possible cases:
> 
> * only happening on debug build
> * adhoc apk above is not built correctly
> * not caused by this PR

when building, did you delete your `node_modules` and run a fresh `npm i` before?

**Comment by chrispader** [comment]
@situchan i just deleted `node_modules`, ran `npm i` and built Android from scratch, and i don't see this issue:

Screenshot_1706263052

**Comment by chrispader** [comment]
The problem you're experiencing is most likely to the patch (that was removed in this PR) still being applied.

I might want to post a Slack message to `#expensify-open-source` to remove `node_modules` and re-install, if anyone is experiencing this.

Could you check if that solution works for you? @situchan

**Comment by situchan** [comment]
> @situchan i just deleted `node_modules`, ran `npm i` and built Android from scratch, and i don't see this issue:
> 
> Screenshot_1706263052

Same worked for me. Sorry for confusion.
I believe many devs will report this in slack.

**Comment by situchan** [comment]
It took 15 min to build android. Much faster 👍

**Comment by situchan** [comment]
> I might want to post a Slack message to #expensify-open-source to remove node_modules and re-install, if anyone is experiencing this.

Agree

**Pull request merged**
